tests/Instances/Data/VectorNonEmpty.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-|
+Module:      Instances.Data.Vector
+Copyright:   (C) 2014-2017 Ryan Scott
+License:     BSD-style (see the file LICENSE)
+Maintainer:  Ryan Scott
+Stability:   Provisional
+Portability: GHC
+
+Provides 'Arbitrary' instances for 'Vector' types.
+-}
+module Instances.Data.VectorNonEmpty () where
+
+import           Data.Vector.NonEmpty as B (NonEmptyVector, fromList)
+
+import           Prelude ()
+import           Prelude.Compat
+
+import           Test.QuickCheck (Arbitrary(..), listOf1)
+import           Test.QuickCheck.Instances ()
+import           Data.Maybe (fromJust)
+
+instance (Arbitrary a) => Arbitrary (B.NonEmptyVector a) where
+    arbitrary = (fromJust . B.fromList) <$> (listOf1 arbitrary)
```

Comment:
I'm not a fan of this use of `fromJust`. (I see why it works here, but I'd prefer to avoid having to use a partial function if possible.) How about doing something like this instead?

```hs
  arbitrary = fromNonEmpty <$> arbitrary
```

Using the [`fromNonEmpty`](https://hackage.haskell.org/package/nonempty-vector-0.2.2.0/docs/Data-Vector-NonEmpty.html#t:NonEmptyVector) function from `nonempty-vector`.